core/src/letter_distribution.c

```diff
@@ -51,26 +51,33 @@ void load_letter_distribution(LetterDistribution *letter_distribution,
       (uint32_t *)malloc(letter_distribution->size * sizeof(uint32_t));
   letter_distribution->is_vowel =
       (uint32_t *)malloc(letter_distribution->size * sizeof(uint32_t));
-  letter_distribution->human_readable_letter_to_machine_letter =
-      (uint32_t *)malloc((MACHINE_LETTER_MAX_VALUE + 1) * sizeof(uint32_t));
-  letter_distribution->machine_letter_to_human_readable_letter =
-      (uint32_t *)malloc((MACHINE_LETTER_MAX_VALUE + 1) * sizeof(uint32_t));
+
+  for (int i = 0; i < MACHINE_LETTER_MAX_VALUE; i++) {
+    letter_distribution->machine_letter_to_human_readable_letter[i][0] = '\0';
+  }
 
   int machine_letter = 0;
   char line[100];
+  int max_tile_length = 0;
   while (fgets(line, sizeof(line), file)) {
-    // For now, we assume 1 char == 1 letter
     // This does not hold true for all languages
     // and will have to be updated.
```

Comment:
These also need to be deleted (it was not obvious they were part of the same comment).